clients/flash/air-client/src/org/bigbluebutton/air/poll/views/PollPopUpMediator.as

```diff
@@ -24,25 +26,30 @@ package org.bigbluebutton.air.poll.views {
 		public override function initialize():void {
 			meetingData.polls.pollChangeSignal.add(onPollChange);
 			view.addEventListener(MouseEvent.CLICK, viewMouseEventHandler);
+			
+			view.removeAllElements();
+			view.addButtons(meetingData.polls.getCurrentPoll());
+			view.stage.addEventListener(StageOrientationEvent.ORIENTATION_CHANGE, onStageOrientation);
 		}
 		
 		private function viewMouseEventHandler(event:MouseEvent):void {
-			if (event.target is Button) {
-				view.removeAllElements();
+			if (event.target == view.closeButton) {
+				view.close();
+			} else if (event.target is Button) {
 				respondToPollSignal.dispatch(event.target.name);
+				view.close();
 			}
 		}
 		
+		private function onStageOrientation(event:StageOrientationEvent):void {
+			view.addButtons(meetingData.polls.getCurrentPoll());

```

Comment:
Why do you re-add the poll options on orientation change?

Reply:
The window size changes depending on the orientation, we need to add buttons back if the popup goes larger. The  `addButtons` removes buttons then add them back.